content/doc/developer/plugin-development/incrementals.adoc

```diff
@@ -0,0 +1,108 @@
+---
+layout: developersection
+title: "Developing plugins and Jenkins core in parallel"
+tags:
+- jenkins
+- incrementals
+- core
+- plugin
+description: "This article explains how to use Incrementals for developing plugins in parallel with Jenkins core"
+---
+
+A common problem one might stumble upon is developing an API in Jenkins core and simultaneously
+building a reference implementation inside a plugin.
+What to do, one might ask.
+
+Worry not, Incrementals are here to the rescue!
+
+With incrementals, we can create Jenkins core incremental versions which can be used along with the plugin to build the
+plugin against the very latest changes you have made to core.
+Also, it allows keeping track and switching between incremental versions.
+Not only that, it allows others to also not have to build Jenkins locally to try out the changes or collaborate with
+you on your awesome project!
+
+Sounds cool? Let's see how to go about it.
+
+== 1. Create a Pull Request in Jenkins core
+
+First step is to link:https://github.com/jenkinsci/jenkins/pulls[create a PR on github] by clicking the "New Pull
+Request" button and push the changes to Jenkins core.
+
+image::/images/developer/plugin-development/incrementals/create-pr.png[]
+
+Note: Make sure that you rebase your branch with the latest changes from master, otherwise incrementals version
```

Comment:
You can simply `git merge` the origin `master` branch; it is necessary to rebase.

Reply:
Agreed with @jglick that this `Note` can be deleted.  Any commit to the branch (either by merge or by new changes) will allow a new incremental version to be delivered.